src/Request.ts

```diff
@@ -0,0 +1,117 @@
+import https from 'https'
+import http from 'http'
+import concat from 'concat-stream'
+import url from 'url'
+import FormData from 'form-data'
+import { UploadError, UploadErrorCode } from './UploadError'
+
+type SourceMapPayload = BrowserSourceMapPayload
+
+export interface BrowserSourceMapPayload {
+  apiKey: string
+  appVersion?: string
+  minifiedUrl: string
+  sourceMap: { filepath: string, data: string }
+  minifiedFile?: { filepath: string, data: string }
+  overwrite?: boolean
+}
+
+const MAX_ATTEMPTS = 5
+const RETRY_INTERVAL = parseInt(process.env.BUGSNAG_RETRY_INTERVAL as string) || 1000
+const TIMEOUT = parseInt(process.env.BUGSNAG_TIMEOUT as string) || 30000
```

Comment:
A comment mentioning these are in ms might be helpful, or maybe rename them to `RETRY_INTERVAL_MS`/`TIMEOUT_MS` ?